WordPress/Classes/ViewRelated/Me/My Profile/Gravatar/GravatarQuickEditorPresenter.swift

```diff
@@ -0,0 +1,46 @@
+import Foundation
+import GravatarUI
+import WordPressShared
+import WordPressAuthenticator
+
+@MainActor
+struct GravatarQuickEditorPresenter {
+    let email: String
+    let authToken: String
+
+    init?(email: String) {
+        let context = ContextManager.sharedInstance().mainContext
+        guard let account = try? WPAccount.lookupDefaultWordPressComAccount(in: context) else {
+            return nil
+        }
+        self.email = email
+        self.authToken = account.authToken
+    }
+
+    func presentQuickEditor(on presentingViewController: UIViewController) {
+        let presenter = QuickEditorPresenter(
+            email: Email(email),
+            scope: .avatarPicker(AvatarPickerConfiguration(contentLayout: .horizontal())),
+            configuration: .init(
+                interfaceStyle: presentingViewController.traitCollection.userInterfaceStyle
+            ),
+            token: authToken
+        )
+        presenter.present(
+            in: presentingViewController,
+            onAvatarUpdated: {
+                AuthenticatorAnalyticsTracker.shared.track(click: .selectAvatar)
+                Task {
+                    // Purge the cache otherwise the old avatars remain around.
+                    await ImageDownloader.shared.clearURLSessionCache()
+                    await ImageDownloader.shared.clearMemoryCache()
```

Comment:
Purging all cache entries seems a bit aggressive. Would only removing the cached image for `email` work?